### PR TITLE
Homework week 4 Burak Özmen

### PR DESCRIPTION
**Pull request by brkzmn**

### CODE REVIEW

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js [hunk 1]
```diff
@@ -11,14 +11,9 @@ Let's rewrite it (or _refactor_ it, as experienced developers would call it):
...
-  const newNumbers = [];
-  for (let i = 0; i < numbers.length; i++) {
-    if (numbers[i] % 2 === 0) {
-      newNumbers.push(numbers[i] * 2);
-    }
-  }
-  return newNumbers;
+  return numbers
+    .filter((number) => number % 2 === 0)
+    .map((number) => number * 2);
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:16]: 👍

File: 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js [hunk 2]
```diff
@@ -31,8 +31,15 @@ const mondayTasks = [
...
-  // TODO complete this function
+function computeEarnings(tasks, ratesPerHour) {
+  const taskEarnings = tasks.map((task) => {
+    return (task.duration / 60) * ratesPerHour;
+  });
+  const totalEarning = taskEarnings.reduce((total, earning) => {
+    return total + earning;
+  }, 0);
+  const fixedEarning = totalEarning.toFixed(2);
+  return `€${fixedEarning}`;
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js:42]: 👍

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 3]
```diff
@@ -26,26 +26,26 @@ const fruitBasket = [
 ];
 
 // ! Function under test
-function sanitizeFruitBasket(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function sanitizeFruitBasket(basket, allergicFruit) {
+  return basket.filter((fruit) => fruit !== allergicFruit);
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:30]: 👍

File: 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js [hunk 4]
```diff
@@ -17,10 +17,12 @@ function createObservable() {
...
   return {
     subscribe: function (subscriber) {
-      // TODO complete this function
+      subscribers.push(subscriber);
     },
     notify: function (message) {
-      // TODO complete this function
+      subscribers.forEach((listener) => {
+        listener(message);
+      });
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js:25]: 👍

**Review by yash-kapila — APPROVED** [review]
Hi @brkzmn - I've finished reviewing the homework. Good work. However, please take care about pushing files to Git which are not necessary. For example, out of the 19 files changed, only 5 of them were necessary.

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js (hunk 1 above)
Comment (brkzmn) [reply 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:16]: Hi @yash-kapila, thank you for your review. Next time, I will take this into account.

**Pull request closed without merging**